lib/ipam/ipam_test.go

```diff
@@ -401,6 +427,34 @@ var _ = testutils.E2eDatastoreDescribe("IPAM tests", testutils.DatastoreEtcdV3,
 		})
 	})
 
+	Describe("IPAM AutoAssign using ip pool node selectors", func() {
+		It("should favor node ip pool with nodeSelector foo=bar", func() {
+			host := "host"
+			pool1 := cnet.MustParseNetwork("10.0.0.0/24")
+			pool2 := cnet.MustParseNetwork("20.0.0.0/24")
+
+			bc.Clean()
+			deleteAllPools()
+
+			applyNode(bc, host, nil)
+			applyPool(pool1.String(), true, `foo == "bar"`)
```

Comment:
I suspect if you applied the selector to the other pool, then this test would start failing (because of the bug mentioned above w.r.t determining pools).

To test this, let's enhance this testcase to test allocations on two nodes - one which is matched by pool1, and one for pool2.

Reply:
(e.g., add selector `foo != bar`)

Reply:
addressed 9fe3116